tests/test_bears/LineCountTestBear.py

```diff
@@ -13,6 +13,6 @@ def run(self, filename, file):
         """
         yield Result.from_values(
             origin=self,
-            message='This file has {count} lines.'.format(count=len(file)),
+            message='This file has {count} lines.'.format(count=len(file.lines)),
```

Comment:
So the FileFactories are all collected in `get_file_dict` and then they are passed around in the bears for analysis right. So before this the bears used to receive the file-contents.

Can you explain this to me a little better. What would the bears receive now to run the analysis on whether it is `FileFactory` or `FileFactory.lines`? I don't know all the details of a coala run. 

Is there a part of the codebase where we can make changes so that all the bears receive only the file contents and not the objects so that we don't have to change the bears implementation individually, you know like a middleware between the collection of objects and the actual analysis?

Reply:
> Can you explain this to me a little better? What would the bears receive now to run the analysis on whether it is FileFactory or FileFactory.lines? I don't know all the details of a coala run.

Definitely `FileFactory.lines` for now. The old bears expect that currently.

> Is there a part of the codebase where we can make changes so that all the bears receive only the file contents and not the objects so that we don't have to change the bears implementation individually, you know like a middleware between the collection of objects and the actual analysis?

Yeah inside Processing.py. Instead of giving bears the file factory objects, give them `file_factory.lines`. I know we currently loose the lazy-loading capability with that, but that's easiest for now.

If you want to keep the lazy loading capabilities, we could write (like you suggested) some kind of "middleware", meaning we have a proxy dict object that gives access to file contents that works the same way as before for bears, but calls the right functions of `FileFactory` in the background.
Like:
```python
class FileDict:  # mimics dict
    def __getitem__(self, filename):
        return self._my_file_middlewares[filename]
    #...

class FileMiddleware:
    def __getitem__(self, lineindex):
        return self._underlying_file_proxy.lines[lineindex]
    #...
```